refactor: s2 icon updates

castastrophe opened the pull request:
## Description

Introducing the Spectrum 2 workflow icons! 🎉 

## Motivation and Context

With the rolling out of Spectrum 2's new designs, a new set of iconography has been created and with that, some improvements on how those icons are provided.  To adjust for this, breaking changes have been made to this repository's architecture in order to support providing iconography in it's new format:

**Spectrum CSS Workflow icons**

Found in `./dist/`, this folder contains the contents published to npm for `@adobe/spectrum-css-workflow-icons`. This package represents S2 iconography and will now ship 3 formats: components (i.e., web components), react, and svg. At the root of this package, you can find a manifest (`dist/manifest.json`) in JSON format of all available assets, as well as a prebuilt spritesheet for the SVGs called `assets/svg-spriteSheet/icons.svg`.

_Note_: Providing a PR preview link was out of scope for the work being done here but is planned for a future iteration.

## How Has This Been Tested?

### Setup

- Check out this repository locally
- `yarn install`
- To preview SVG icons: `yarn serve`

**TODO:** Need to find and summarize the changed icons and list out any deprecated and new icons.

## Screenshots

<img width="2238" alt="Screenshot 2024-05-02 at 9 39 04 AM" src="https://github.com/adobe/spectrum-css-workflow-icons/assets/1840295/1b518934-e728-4417-ac22-a31fbfee7b85">

## Types of changes

- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.

karstens commented:
@castastrophe I really like what you are doing, thank you so much! How can I test this without publishing anything? Just check out the branch and build it and manually look/check the `dist` folder?

@Hinschmi @jianliao Are you okay with this?

jianliao commented:
@castastrophe ~~Do I need yarn 4 for this project?~~

Never mind. It is yarn v1, I figure it out from the `yarn.lock` file.

pfulton reviewed (approved):
Thanks for taking care of this huge effort. After getting Artifactory access configured correctly, I was able to follow the steps that you outlined in the description, ran the scripts, and verified that the icons looked like they should in the browser.

jianliao commented:
LGTM

karstens commented on README.md:
```diff
@@ -57,11 +54,9 @@ The following tasks are available:
-| `build` | Cleans the dist directory, copies website assets and then brings in the rsp, express, and workflow icons into dist | `yarn build` |
-| `build:icons` | Deletes and rebuilds the icon folder, can only be run by an Adobe employee with access | `yarn build:icons` |
-| `build:site` | Cleans the dist directory, copies website assets and prepares the dist output for the site preview | `yarn build:site` |
+| `build` | Deletes and rebuilds the icon folder, can only be run by an Adobe employee with access | `yarn build:icons` |
```
@castastrophe Little typo, I guess. The command at the end of the line is just `yarn build`.

castastrophe replied:
Great catch thank you!

karstens reviewed (approved):
Did not look at the changeset or git workflows, but the asset handling and tasks do look good to me. 👍

castastrophe commented on .github/workflows/build.yml:
```diff
@@ -0,0 +1,131 @@
+                  enable-pull-request-comment: true
+                  enable-commit-comment: false
+                  overwrites-pull-request-comment: true
+                  alias: pr-${{ github.event.number }}
```
```suggestion
                  alias: deploy-preview-${{ github.event.number }}
```

castastrophe committed:
Major refactor workflow icons to Spectrum 2

** Breaking change **

Icon sets have been refactored to provide a more consistent and maintainable set of icons.

- No longer provided in 3 different formats: 18 (small), 24 (large), and color
- All icons provided in 20x20 viewboxes except where noted (i.e., 22x20)
- No "color" sets, however all icons are shipped with custom property hooks

See the MIGRATION.md document for detailed information on how to migrate to the new set.
